components/collection/activity/events/eventRow/common.ts

```diff
@@ -0,0 +1,79 @@
+import { Interaction } from '@kodadot1/minimark'
+import {
+  InteractionWithNFT,
+  Offer,
+  OfferInteraction,
+} from '@/components/collection/utils/types'
+import { sanitizeIpfsUrl } from '~~/utils/ipfs'
+import { processSingleMetadata } from '~~/utils/cachingStrategy'
+import { NFTMetadata } from '~~/components/rmrk/service/scheme'
+
+export const ineteractionNameMap = {
+  BUY: 'Sale',
+  LIST: 'List',
+  MINTNFT: 'Mint',
+  SEND: 'Transfer',
+  Offer: 'Offer',
+}
+export const blank = '--'
+
+export const interactionColor = {
+  [Interaction.MINTNFT]: 'k-yellow',
+  [Interaction.LIST]: 'k-blueaccent',
+  [Interaction.BUY]: 'k-pink',
+  [OfferInteraction]: 'k-greenaccent',
+  [Interaction.SEND]: 'background-color',
+}
+
+export const getAmount = (
+  event: InteractionWithNFT | Offer
+): string | number => {
+  switch (event.interaction) {
+    case Interaction.MINT:
+      return blank
+    case Interaction.LIST:
+    case Interaction.BUY:
+      return (event as InteractionWithNFT).meta
+    case OfferInteraction:
+      return (event as Offer).price
+    default:
+      return blank
+  }
+}
+
+export const getFromAddress = (event: InteractionWithNFT | Offer): string => {
+  const interaction = event.interaction
+  if (interaction === Interaction.MINTNFT || interaction === Interaction.LIST) {
+    return blank
+  }
+  if (interaction === Interaction.BUY || interaction === Interaction.SEND) {
+    return (event as InteractionWithNFT).currentOwner
+  }
+  if (interaction === OfferInteraction) {
+    return (event as Offer).caller
+  }
+  return blank
+}
+
+export const getToAddress = (event: InteractionWithNFT | Offer): string => {
+  const interaction = event.interaction
+  if (interaction === Interaction.MINTNFT || interaction === Interaction.LIST) {
+    return event.caller
+  }
+  if (interaction === Interaction.BUY || interaction === Interaction.SEND) {
+    return event.caller
```

Comment:
should it return `caller` for the send event? I guess not

Reply:
correct. thank you. it should be meta

Reply:
Done. thank you

Reply:
Then we could put MINTNFT LIST BUY events in the same condition. I would prefer to use `switch case` statement

Reply:
noted.
with you permission let's handle in follow up

Reply:
It's a small change. At least we could write so please
```
if (interaction === Interaction.MINTNFT || interaction === Interaction.LIST || interaction === Interaction.BUY)  
{ 
xxx
}
```

Reply:
sure. ok.  will do

Reply:
done